src/components/videoCard.jsx

```diff
@@ -10,33 +12,86 @@ export function VideoCard({ video }) {
     author,
     likes,
     views,
+    videoId,
   } = video;
+  const {
+    resourcesApiDispatch,
+    userResources: { likes: likedVideos, watchlater },
+  } = useUserResources();
+  const {
+    ADD_TO_LIKES,
+    ADD_TO_WATCHLATER,
+    REMOVE_FROM_WATCHLATER,
+    REMOVE_FROM_LIKES,
+  } = resourcesApiStateEnums;
+
+  const {
+    userState: { isLoggedIn },
+    setShowAuthModal,
+  } = useAuth();
+
+  const isLiked = likedVideos.some((video) => video._id === id);
+  const isAddedToWatchlater = watchlater.some((video) => video._id === id);
+
```

Comment:
just a suggestion , can make isIteminArray separate function for this

Reply:
Thanks for the suggestion, will try to implement it